### DISCUSSION TITLE
How to render screenshots offscreen and save as image?

### DISCUSSION THREAD

**Discussion by lammermann** [emilk/egui#4432] [Q&A] [answered]
I try to create a gui with egui/eframe. For testing purposes I would like to give my application a state and then save the gui as an image instead of showing it.

At the moment I use the __screenshot feature of eframe, but this always pops up the window shortly and thus hurts my programming flow. Is it possible render directly to an image?

Any help is highly appreciated.

**Comment by YgorSouza** [comment]
This is the goal of #1129, but it is not implemented yet. See also #3926 and <https://github.com/lucasmerlin/egui_skia>.

**Comment by lammermann** [comment, marked as answer]
With the newest version of egui this issue is solved by https://github.com/emilk/egui/tree/master/crates/egui_kittest
